### PR TITLE
product_customer_code migrated to 8.0. Add search by customer code and name.

### PR DESCRIPTION
**Pull request by acysos**

### CODE REVIEW

File: product_customer_code/models/product.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+##############################################################################
+from openerp import models, fields, api
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    product_customer_code_ids = fields.One2many(
+        comodel_name='product.customer.code', inverse_name='product_id',
+        string='Customer Codes')
```
Comment (pedrobaeza) [inline product_customer_code/models/product.py]: s/Codes/Code

Comment (pedrobaeza) [inline product_customer_code/models/product.py]: Don't matter. I didn't see it was a one2many.

File: product_customer_code/models/product.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                product_ids.update(product_customer_code_obj.search(
+                    cr, uid, [('product_name', operator, term)], limit=limit2,
+                    context=context))
+            product_ids = list(product_ids)
+
+            product_datas = product_customer_code_obj.browse(cr, uid,
+                                                             product_ids,
+                                                             context)
+
+            prod_ids = []
```
Comment (pedrobaeza) [inline product_customer_code/models/product.py:72]: You can do: `res = [x.product_id.id for x in product_datas]`

File: product_customer_code/models/product.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+               context=None, count=False):
+        res = super(ProductTemplate, self).search(cr, uid, args, offset=offset,
+                                                  limit=limit, order=order,
+                                                  count=count)
+        operator = False
+        term = False
+        for arg in args:
+            if arg[0] == 'name':
+                operator = arg[1]
+                term = arg[2]
```
Comment (pedrobaeza) [inline product_customer_code/models/product.py:55]: `break`

File: product_customer_code/models/product.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                                                 limit=limit, order=order,
+                                                 context=context, count=count)
+        operator = False
+        term = False
+        for arg in args:
+            if arg[0] == 'name':
+                operator = arg[1]
+                term = arg[2]
+
+        if not res and operator is not False and term is not False:
```
Comment (pedrobaeza) [inline product_customer_code/models/product.py:95]: Can't you call it here template search?

**Comment by pedrobaeza** [comment]
@acysos, any news on this?

**Comment by acysos** [comment]
It's a pending task, I have to revise the changes that suggested.

**Comment by gurneyalex** [comment]
needs rebasing.

**Comment by hurrinico** [comment]
In order to rebase this PR i've created https://github.com/OCA/product-attribute/pull/122

**Comment by eLBati** [comment]
So, this should be closed in favor of https://github.com/OCA/product-attribute/pull/122 right?

**Comment by hurrinico** [comment]
@eLBati IMHO yes

**Pull request closed without merging**